src/netconf_subscribed_notifications.c

```diff
@@ -152,49 +152,39 @@ sub_ntf_find_next(struct np2srv_sub_ntf *last, int (*sub_ntf_match_cb)(struct np
     return NULL;
 }
 
-int
-csn_send_notif(struct csn_receiver_info *recv_info, uint32_t nc_sub_id,
-        struct timespec timestamp, struct lyd_node **ly_ntf, int use_ntf)
+/**
+ * @brief Build a notification.
+ *
+ * @param[in] timestamp any timestamp, mainly the current time.
+ * @param[in] ly_ntf the notification content to add to the message.
+ * @return a pointer to a new unyte message on success, NULL on failure.
+ */
+static unyte_message_t *
+csn_build_notification(struct timespec timestamp, struct lyd_node **ly_ntf)
 {
-    char *string_notification = NULL;
     unyte_message_t *message = NULL;
     static uint32_t message_id = 0;
     char *string_to_send = NULL;
-    struct np2srv_sub_ntf *sub;
     char *eventtime = NULL;
-    int rc = SR_ERR_OK;
-    uint32_t r;
 
-    /* find the subscription structure */
     ly_time_ts2str(&timestamp, &eventtime);
+    message = (unyte_message_t *)malloc(sizeof *message);
+    if (!message) {
+        goto cleanup;
+    }
+
     lyd_print_mem(&string_to_send, *ly_ntf, LYD_XML, LYD_PRINT_WD_ALL | LY_PRINT_SHRINK);
-    if (asprintf(&string_notification,
+    if (asprintf((char **)&message->buffer,
             "<notification xmlns:\""NC_NS_NOTIF "\">"
             "<eventTime>%s</eventTime>"
             "%s"
             "</notification>",
             eventtime, string_to_send) < 0) {
-        EINT;
-        rc = SR_ERR_NO_MEMORY;
-        goto cleanup;
-    }
-
-    sub = sub_ntf_find(nc_sub_id, 0, 0, 0);
-    if (!sub) {
-        EINT;
-        rc = SR_ERR_INTERNAL;
-        goto cleanup;
-    }
-
-    message = (unyte_message_t *)malloc(sizeof *message);
-    if (!message) {
-        EINT;
-        rc = SR_ERR_NO_MEMORY;
+        free(message);
         goto cleanup;
     }
 
-    message->buffer = string_notification;
-    message->buffer_len = strlen(string_notification);
+    message->buffer_len = strlen(message->buffer);
```

Comment:
Minor improvement, the message length is returned by `asprintf()` so it saves you this `strlen()` call.